### PR TITLE
feat(cop_dataspace): adds S1_SAR_GRD_COG and new odata query parameters

### PR DESCRIPTION
**Pull request by ninsbl**
### Description
The `cop_dataspace` provider uses COG-format based GRD data as main target now. This PR adds the `S1_SAR_GRD_COG` product type and enhances querying possibilities in `cop_dataspace` allowing to query based on `publicationDate` and `modificationDate` using terms:

- `modifiedAfter`
- `modifiedBefore`
- `publishedAfter`
- `publishedBefore`

It also fixes `orbitNumber` and `relativeOrbitNumbe` searches, that either were ignored or threw errors.
 
### Further comments
Filtering by publishing or modification date might relevant for other providers too...

### CODE REVIEW

**Comment by ninsbl** [comment]
Hmm... `python /opt/src/eodag/utils/params_mapping_to_csv.py` fails when I run tests locally, It seems to write windows line endings and then CSV writer fails with:
`UnicodeEncodeError: 'latin-1' codec can't encode character '\u2019' in position 131: ordinal not in range(256)` 

Any idea how to address that?

**Comment by ninsbl** [comment]
The test-failure here: https://github.com/CS-SI/eodag/actions/runs/10109344749/job/27957248216?pr=1277 seems unrelated to the changes...

**Comment by sbrunato** [comment]
Hello @ninsbl , and many thanks for this PR!

> The test-failure here: https://github.com/CS-SI/eodag/actions/runs/10109344749/job/27957248216?pr=1277 seems unrelated to the changes...

This is because the PR come from a fork and does not have permissions to write tests results in the PR through the dedicated GH-action. No problem, as we can see that they succeeded

**Comment by sbrunato** [comment]
> Hmm... `python /opt/src/eodag/utils/params_mapping_to_csv.py` fails when I run tests locally, It seems to write windows line endings and then CSV writer fails with: `UnicodeEncodeError: 'latin-1' codec can't encode character '\u2019' in position 131: ordinal not in range(256)`
> 
> Any idea how to address that?

it is ok, I just added remaining product information generated with `tox -e docs` and included it in the PR

File: eodag/resources/product_types.yml [hunk 1]
```diff
@@ -554,6 +554,30 @@ S1_SAR_GRD_JP2:
...
+    projected to ground range using an Earth ellipsoid model. Phase information is lost. The resulting product has
+    approximately square spatial resolution pixels and square pixel spacing with reduced speckle at the cost of worse
+    spatial resolution.
+    GRD products can be in one of three resolutions: |
+      Full Resolution (FR),
+      High Resolution (HR),
+      Medium Resolution (MR).
+    The resolution is dependent upon the amount of multi-looking performed. Level-1 GRD products are available in MR
+    and HR for IW and EW modes, MR for WV mode and MR, HR and FR for SM mode.
+    Product without SAFE formatting, see
```
Comment (sbrunato) [inline eodag/resources/product_types.yml:569]:
```suggestion
    Product containing Cloud Optimized GeoTIFF images, without SAFE formatting.
```

**Review by sbrunato — CHANGES_REQUESTED** [review]

**Review by sbrunato — APPROVED** [review]

**Pull request merged**